### PR TITLE
feat(test): enhance backend integration test sequence tracking and helpers

### LINKED ISSUE #4319: Enhance backend integration test infrastructure (sequence resync, test helpers, config cleanup)
Improve the reliability and ergonomics of backend integration tests by fixing PostgreSQL sequence drift, adding reusable test helper methods, and cleaning up duplicate bean definitions in the test config.

### 1. Automatic PostgreSQL Sequence Resynchronization
- [ ] In `BaseWebContextSensitiveTest.java`, extend `FIXTURE_SEQUENCE_MAPPINGS` to include: `organization`, `analyzer`, `referral_status_history`, `calculation`, `result_limits`, `site_information`.
- [ ] Verify DBUnit fixture loads automatically resync sequence `setval` to `MAX(id) + 1` for all populated entity tables.
- [ ] Confirm this prevents `duplicate key value` constraint violations during test runs.

### 2. Test Authoring Helper Methods
- [ ] Add `resyncAllSequences()` to `BaseWebContextSensitiveTest.java` — resyncs all entity sequences in one call for multi-entity programmatic inserts.
- [ ] Add `performGet(String url)` — MockMvc GET helper with JSON headers.
- [ ] Add `performPost(String url, Object content)` — MockMvc POST helper with JSON payload serialization and headers.

### 3. Clean Configuration & Bean Wiring
- [ ] In `AppTestConfig.java`, remove duplicate mock bean definitions for `RequesterTypeService` and `OrganizationTypeService` that conflict with component-scanned services under `@ActiveProfiles("test")`.
- [ ] Remove unused imports in `AppTestConfig.java`.

### PR DESCRIPTION
**Pull request by Agaba-derrick**
# Pull Requests Requirements

- [ ] The PR title includes a brief description of the work done, including the
      Issue number if applicable.
- [ ] The PR includes a video showing the changes for the work done.
- [ ] The PR title follows conventional commit label standards.
- [ ] The changes confirm to the OpenElis Global x3
      [Styleguide and Design](https://uwdigi.atlassian.net/wiki/spaces/OG/pages/621346838/OpenELIS+Global+Style+Guide)
      documentation.
- [ ] The changes include tests or are validated by existing tests.
- [ ] I have read and agree to the Contributing
      [Guidelines](https://openelis-global.org/community/get-involved/) of this
      project.

## Summary
<p>This PR implements and verifies backend integration test infrastructure enhancements on <code>feat/backend-integration-test-enhancements</code>. It fixes PostgreSQL sequence drift during DBUnit fixture loads, adds reusable test authoring helpers, and cleans up duplicate bean definitions in the test config.</p>
<h2>Changes</h2>
<h3>1. Automatic PostgreSQL Sequence Resynchronization</h3>
<p><strong>File:</strong> <code>src/test/java/org/openelisglobal/BaseWebContextSensitiveTest.java</code> (L108–115)</p>
<ul>
<li>Added <code>organization</code>, <code>analyzer</code>, <code>referral_status_history</code>, <code>calculation</code>, <code>result_limits</code>, and <code>site_information</code> sequences to <code>FIXTURE_SEQUENCE_MAPPINGS</code>.</li>
<li>DBUnit fixture loads now automatically resync sequence <code>setval</code>s to <code>MAX(id) + 1</code> for all populated entity tables, preventing PostgreSQL <code>duplicate key value</code> constraint violations.</li>
</ul>
<h3>2. Test Authoring Helper Methods</h3>
<p><strong>File:</strong> <code>src/test/java/org/openelisglobal/BaseWebContextSensitiveTest.java</code> (L585–623)</p>
<p>Added ergonomic JUnit 4 helper methods:</p>
<ul>
<li><code>resyncAllSequences()</code> — resyncs all entity sequences in a single call for multi-entity programmatic inserts.</li>
<li><code>performGet(String url)</code> — helper for MockMvc GET requests with JSON headers.</li>
<li><code>performPost(String url, Object content)</code> — helper for MockMvc POST requests with JSON payload serialization and headers.</li>
</ul>
<h3>3. Clean Configuration &amp; Bean Wiring</h3>
<p><strong>File:</strong> <code>src/test/java/org/openelisglobal/AppTestConfig.java</code> (L373–380)</p>
<ul>
<li>Cleaned up duplicate mock bean definitions (<code>RequesterTypeService</code>, <code>OrganizationTypeService</code>) that conflicted with component-scanned services when <code>@ActiveProfiles("test")</code> is active.</li>
<li>Removed unused imports.</li>
</ul>
<h2>Testing</h2>
<h3>Automated Test Suite Runs</h3>

Test | Command | Result
-- | -- | --
PatientDAORedirectIntegrationTest | mvn test -Dtest=PatientDAORedirectIntegrationTest | Tests run: 8, Failures: 0, Errors: 0, Skipped: 0 — PASS
CytologySampleServiceTest | mvn test -Dtest=CytologySampleServiceTest | Tests run: 28, Failures: 0, Errors: 0, Skipped: 0 — PASS

<h2>Notes for Reviewers</h2>
<ul>
<li>No production code was touched — all changes are scoped to the test infrastructure (<code>BaseWebContextSensitiveTest.java</code>, <code>AppTestConfig.java</code>).</li>
<li>The sequence resync logic addresses a recurring flakiness source in fixture-based integration tests and should reduce false-negative failures caused by stale sequence state.</li>
</ul></body></html>
## Related Issue

Closes #4319

### CODE REVIEW

File: src/test/java/org/openelisglobal/BaseWebContextSensitiveTest.java [hunk 1]
```diff
@@ -577,4 +580,45 @@ protected void ensureSiteInformation(String name, String value) {
...
         }
     }
+
+    /**
+     * Resync all tracked entity sequences to MAX(id)+1. Convenience method for
+     * tests that insert multiple records programmatically across different
+     * entities.
+     */
+    protected void resyncAllSequences() {
+        try (Connection conn = dataSource.getConnection()) {
```
Comment (HerbertYiga) [inline src/test/java/org/openelisglobal/BaseWebContextSensitiveTest.java]: conn is declared but I don't see it referenced anywhere,  Was it meant to be passed into the loop?

Comment (Agaba-derrick) [reply src/test/java/org/openelisglobal/BaseWebContextSensitiveTest.java]: @HerbertYiga I have refactored  BaseWebContextSensitiveTest.java to address 
Overloaded resyncSequence(Connection conn, String sequence, String table): Allows callers to pass an existing connection.
Updated resyncSequence(String sequence, String table): Preserves backward compatibility for other test callers by delegating to resyncSequence(conn, ...).
Updated resyncAllSequences(): Reuses the single acquired conn for all mapping sequence resyncs inside the loop.

**Comment by pmanko** [comment]
@Agaba-derrick maintainer note: we've extended this PR because it's the right home for fixing the backend flake that's been turning `develop` red.

**Why develop kept going red.** Every backend integration test shares one Postgres container, and nothing gets rolled back between classes (`BaseWebContextSensitiveTest` is `@Transactional(NOT_SUPPORTED)`). Each class therefore sees whatever state the classes before it left behind. Surefire had no `runOrder` set, so it fell back to `filesystem` order, which is fixed per GitHub runner image. On `ubuntu24/20260920.314` that order fails 33 tests every time; on `20260907.300` it passes. Commit `c457531524` failed on one image and passed on the other.

**What's been added here**
1. `<runOrder>alphabetical</runOrder>` in the surefire config. The order no longer depends on the runner, and failures now reproduce (the same 7 on two separate runs).
2. Fixes for those 7:
   - `reflex_rule`, `reflex_rule_condition` and `reflex_rule_action` added to `FIXTURE_SEQUENCE_MAPPINGS`. `logbook-db.xml` seeds explicit ids, so a later service-created rule collided at id 1.
   - `ResultSelectListOptionsTest` creates its own `test` row instead of borrowing `min(id)`. A `localization` truncate cascades into `test`, which left the borrowed id null.
   - `DisplayListControllerStageFlagsTest` restores the banner-heading `localization` row if missing. The same cascade removes it, and the endpoint throws without it.
3. Your sequence-drift mappings are kept. They're the same fix pattern as the first item above.

**Small cleanups in the next push**
- Keep `resyncAllSequences()`, `performGet()` and `performPost()`. Nothing calls them yet, but 24 call sites already hand-roll the JSON GET/POST pattern, and several tests resync multiple sequences by hand, so they're worth adopting. The MockMvc helpers switch to imports instead of fully qualified names.
- Remove the duplicated `resyncSequence` javadoc.
- Update the PR description to cover the above.

Follow-up, not in this PR: reset the whole database between classes, so a new test can't bring this back.

**Review by pmanko — APPROVED** [review]

**Comment by Agaba-derrick** [comment]
@pmanko  thank you for the changes on this one, it landed really well
